Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -333,18 +319,19 @@
 %% 7. Switch Hadamard block dimension and innerRepetition dimension back
 % For model fitting, we want the PLDs-first-TEs-second order (just like at the beginning) so we need to reorder it again
 
-% This is the original order
-vectorReorderDecoded = 1:size(decodedPWI4D, 4);
-
-% Shape to a matrix with TEs second and all the rest first
-vectorReorderDecoded = reshape(vectorReorderDecoded, size(decodedPWI4D, 4)/nInnerRepetitions, nInnerRepetitions);
+if nInnerRepetitions > 1
+	% This is the original order
+	vectorReorderDecoded = 1:size(decodedPWI4D, 4);
 
-% Flip the two dimensions and make a row vector again
-vectorReorderDecoded = reshape(vectorReorderDecoded', 1, size(decodedPWI4D, 4));
+	% Shape to a matrix with TEs second and all the rest first
```

Comment:
This is weird. I thought we agreed that TEs should be first (they are the innermost dimension, so this is how we want them in BIDS). So BIDS is TE/PLD/n. Because this is the acquisition order.

And for decoding you needed PLD/TE/n, this is how you explained it to me.

Of course we don't care what we need for modeling, that's just a convention by FABBER right?
If needed, we change this in xASL_quant_FSL (this is exactly where that function is for). But throughout the pipeline, and as rawdata, it should be stored as BIDS. And if this is not yet decided in BIDS, then we should go for acquisition order.

Reply:
It's all correct. You understand it correctly and the comment is correct. 

But more description is probably warranted so that you do not mistake this comment for the final results. 

I've updated the comments. Basically it goes as:

You start with a 4D matrix.
Then you first reshape to 5D with all TEs are second (5th dim) and the rest first (4th dim) = this line of code.
 
Then you flip 4D and 5D. And then you make 4D again.